WolvenKit.RED4/Archive/IO/RedPackageReader.cs

```diff
@@ -53,9 +54,9 @@ public override void ReadClass(RedBaseClass cls, uint size)
 
                 var propRedType = RedReflection.GetRedTypeFromCSType(prop.Type, prop.Flags);
 
-                if (propRedType != redTypeName)
+                if (redTypeName != CName.Empty && propRedType != redTypeName)
```

Comment:
Same here, why the additional check?

Reply:
Same reason, empty string is not a valid key here